signac/contrib/collection.py

```diff
@@ -452,15 +615,44 @@ def __getitem__(self, _id):
 
     @staticmethod
     def _validate_key(key):
-        "Emit a warning or raise an exception if key is invalid. Returns key."
+        """Emit a warning or raise an exception if key is invalid. Returns key.
+
+        Parameters
+        ----------
+        key : str
+            Key to validate.
+
+        Returns
+        -------
+        str
+            If key is validated.
+
+        Raises
+        ------
+        InvalidKeyError
+            When key given is invalid.
+
+        """
         if '.' in key:
             from ..errors import InvalidKeyError
             raise InvalidKeyError("Keys may not contain dots ('.').")
         return key
 
     @classmethod
     def _validate_doc(cls, doc):
-        "Emit a warning or raise an exception if the document is invalid. Returns doc."
+        """Emit a warning or raise an exception if the document is invalid. Returns doc.
+
+        Parameters
+        ----------
+        doc : dict
+            The document to validate.
+
+        Returns
+        -------
+        doc : dict
+            If doc is validated.
+
```

Comment:
Missing `Raises` block

Reply:
there is no exceptions raised here, can you please let me know which raise block are you mentioning?

Reply:
Yes, you're right. My bad!